src/main/java/duke/Duke.java

```diff
@@ -0,0 +1,271 @@
+                    if (taskSplit[1].equals("1")) {
+                        t.setDone();
+                    }
+                } else if (taskSplit[0].equals("[E]")) {
+                    Event e = new Event (taskSplit[2], taskSplit[3]);
+                    tasks.add(e);
+                    if (taskSplit[1].equals("1")) {
+                        e.setDone();
+                    }
+                }
+            }
+            readFile.close();
+        } catch (IOException e) {
+            System.out.println("Error when creating data directory. Please try again.");
+        }
+    }
+
+    public static boolean saveData() {
+        try {
+
+            System.out.println("Saving your data...");
+            FileWriter fileWriter = new FileWriter("data/duke.txt");
+            for (int i = 1; i < tasks.size(); i++) {
+                Task currentTask = tasks.get(i);
+                fileWriter.write(currentTask.getTaskType() + " | " + ((currentTask.getIsDone()) ? "1" : "0") + " | "
+                        + currentTask.getDescription() + ((currentTask.hasDateTime()) ? " | "
+                        + currentTask.getDateTime() : "") + "\n" );
+            }
+            fileWriter.close();
+            System.out.println("Data saved");
+            return true;
+        } catch (IOException e) {
+            System.out.println("Error occurred when saving your data. Please try again.");
+            return false;
+        }
+    }
+
+    public static void getInput() {
+        while (true) {
+            Scanner sc = new Scanner(System.in);
+            String inputCommand = sc.nextLine();
+            if (inputCommand.equals("list")) {
+                showList();
+            } else if (inputCommand.equals("bye")) {
+                goodbye();
+                break;
+            } else if (inputCommand.equals("save")) {
+                saveData();
+            } else {
+                try {
+                    String taskToHandle = inputCommand.split(" ", 2)[1];
+                    if (inputCommand.startsWith("done")) {
+                        markAsDone(taskToHandle);
+                    } else if (inputCommand.startsWith("todo")) {
+                        addToDo(taskToHandle);
+                    } else if (inputCommand.startsWith("deadline")) {
+                        addDeadline(taskToHandle);
+                    } else if (inputCommand.startsWith("event")) {
+                        addEvent(taskToHandle);
+                    } else if (inputCommand.startsWith("delete")) {
+                        deleteTask(taskToHandle);
+                    } else {
+                        printInvalidMessage(inputCommand);
+                    }
+                } catch (ArrayIndexOutOfBoundsException e) {
+                    printInvalidMessage(inputCommand);
+                }
+            }
+            System.out.println("_________________________________________________________________________");
```

Comment:
Perhaps refactor the "___________" into a constant to avoid using magic literals.